lib/cn/cn.c

```diff
@@ -497,13 +509,16 @@ cn_mblocks_destroy(
         if (kcompact)
             continue;
         cn_delete_blks(ds, &list[lx].vblks, n_committed);
-    }
-}
 
-static inline size_t
-roundup_size(size_t val, size_t align)
-{
-    return align * ((val + align - 1) / align);
+        /* Same logic as cn_delete_blks() applied to one block. Candidate for
+        * refactor.
+        */
+        if (n_committed > 0) {
+            delete_mblock(ds, &list->hblk);
+        } else {
+            abort_mblock(ds, &list->hblk);
+        }
+    }
```

Comment:
The logic with respect to n_committed seems broken.   And I mean broken from before this PR.  How can one scalar account for committed status for two (and now three) sets of mblocks (kblocks, vblocks and now hblocks)?  And the local var inside cn_delete_blks is decremented, but the caller doesn't know that and uses original value for the next set.  Is it goofy or am I confused?

Reply:
I thought this too, but it looks right to me because all it is saying is that if there were any committed blocks, abort all potential blocks. Mpool understands not to do anything with a 0 blkid. CC @nabeelmmd

Reply:
It still seems fishy.   I'll make a ticket to check it out later.